src/main/java/com/commercetools/sync/categories/CategorySync.java

```diff
@@ -47,15 +48,15 @@
     private final CategoryService categoryService;
     private final CategoryReferenceResolver referenceResolver;
 
-    private final Map<String, List<String>> categoryKeysWithMissingParents = new HashMap<>();
-    private final Set<String> processedCategoryKeys = new HashSet<>();
+    private final Map<String, List<String>> categoryKeysWithMissingParents = new ConcurrentHashMap<>();
+    private final Set<String> processedCategoryKeys = ConcurrentHashMap.newKeySet();
 
     private Set<CategoryDraft> existingCategoryDrafts = new HashSet<>();
     private Set<CategoryDraft> newCategoryDrafts = new HashSet<>();
     private Set<CategoryDraft> referencesResolvedDrafts = new HashSet<>();
-    private Set<String> categoryKeysWithResolvedParents = new HashSet<>();
+    private Set<String> categoryKeysWithResolvedParents = ConcurrentHashMap.newKeySet();
     private Set<String> categoryKeysToFetch = new HashSet<>();
-    private Map<CategoryDraft, Category> categoryDraftsToUpdate = new HashMap<>();
+    private Map<CategoryDraft, Category> categoryDraftsToUpdate = new ConcurrentHashMap<>();
```

Comment:
guess it is better now to divide the declarations int concurrent and non-concurrent groups, otherwise hard to understand which maps/sets must be concurrent, which - not.

Maybe change the field type explicitly to _ConcurrentXXX_ so you can't re-initialize it with unsafe implementation, like in [line 152](https://github.com/commercetools/commercetools-sync-java/blob/51e530543b725d11734f70da9c412100c4b600d9/src/main/java/com/commercetools/sync/categories/CategorySync.java#L152)

Reply:
955d5eb
 Good catch! Thanks a lot.